homeassistant/components/cmus/media_player.py

```diff
@@ -86,11 +86,38 @@ def __init__(self, server, password, port, name):
             self.cmus = remote.PyCmus()
             auto_name = "cmus-local"
         self._name = name or auto_name
+        self._server = server
+        self._password = password
+        self._port = port
         self.status = {}
 
+    def reconnect(self):
+        """Reconnect the CMUS device."""
+
+        try:
+            if self._server:
+                self.cmus = remote.PyCmus(
+                    server=self._server, password=self._password, port=self._port
+                )
+            else:
+                self.cmus = remote.PyCmus()
+        except exceptions.InvalidPassword:
+            _LOGGER.error("The provided password was rejected by cmus")
+            return False
+        except exceptions.CmusNotRunning:
+            _LOGGER.error("cmus does not appear to be running")
+            return False
+
     def update(self):
         """Get the latest data and update the state."""
-        status = self.cmus.get_status_dict()
+        try:
+            status = self.cmus.get_status_dict()
+        except BrokenPipeError:
+            self.reconnect()
+            status = self.cmus.get_status_dict()
+        except exceptions.ConfigurationError:
```

Comment:
When this exception occurs, reconnect is called, but status is not set like in the previous block. I think this will cause `NameError` for `if not status` because `status` is never set.

Reply:
Since we attempt a reconnection here, the status should be accessible if it successfully reconnects.

Though since reconnections are usually from the cmus server being off, I think this might be sensible not to try updating the status here.